Crashlytics/UnitTests/FIRCLSReportManagerTests.m

```diff
@@ -201,17 +203,10 @@ - (NSArray *)uploadReportArray {
 #pragma mark - File/Directory Handling
 - (void)testCreatesNewReportOnStart {
   FBLPromise<NSNumber *> *promise = [self->_reportManager startWithProfiling];
+  FBLWaitForPromisesWithTimeout(1.0);
```

Comment:
This `FBLWaitForPromisesWithTimeout` seems to be more robust then other API for testing purposes. I think we may be able to cut down on flakes by adding this accordingly after more nightly runs. 

An alternative approach to addressing the chronic flakes would probably be to refactor all of the dependencies of this class to offer a testing API that drains the operation/dispatch queues that are used.